ci-operator/step-registry/baremetalds/sno/hostname/baremetalds-sno-hostname-commands.sh

```diff
@@ -41,29 +56,38 @@ wait_approve_csr() {
   oc get csr -o go-template='{{range .items}}{{if not .status}}{{.metadata.name}}{{"\n"}}{{end}}{{end}}' | xargs oc adm certificate approve
 }
 
+wait_for_api
+
 if [[ "$(hostname)" != "another-hostname" ]]
 then
+  echo "Deleting node object before changing the hostname..."
+  oc delete node "$(oc get nodes -ojsonpath='{.items[?(@.metadata.name == "'"$(hostname)"'")].metadata.name}')"
+
   systemctl stop kubelet.service
   # Forcefully remove all pods rather than just stop them, because a different hostname
   # requires new pods to be created by kubelet.
-  crictl rmp --force --all
+  until crictl rmp --force --all &> /dev/null
+  do
+    sleep 2
+  done
   systemctl stop crio.service
+
+  # manually remove multus and OVN client certs, as they are not reconciled after the hostname
+  # change and they contain the hostname in their CN
+  rm -rf /etc/cni/multus/certs/multus-client-*.pem \
```

Comment:
OVN certs are at `/var/lib/ovn-ic/etc` and `/var/lib/openvswitch/pki` too

Reply:
Thanks for the info. I can definitely add the removal of the files under ` /var/lib/openvswitch/pki/*`, although this dir seems to be empty for a baremetal SNO installation. 

The ones we are referring to, were introduced (IIUC) [here](https://github.com/openshift/multus-cni/commit/30558b5e68d3fd316fefa35fa7d258b91773f8a0) and [here](https://github.com/openshift/ovn-kubernetes/commit/7dc48048a346f175772fe5c2839b083f7a20c597). We ended up with these only, because we observed that those are not reconciled unless there are deleted and they contain the node hostname in the CN. Should I just rm files under `/var/lib/openvswitch/pki/*` as well? WDYT?

Reply:
@zshi-redhat ptal

Reply:
Just for reference, the reason we only delete those ones is that, if we don't, then [they won't be reconciled](https://issues.redhat.com/browse/SDN-4220).